staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.pb.go

```diff
@@ -6964,6 +6964,395 @@ func (m *ReopenContainerLogResponse) XXX_DiscardUnknown() {
+	// Options for checkpointing.
+	Options              *CheckpointContainerOptions `protobuf:"bytes,2,opt,name=options,proto3" json:"options,omitempty"`
+	XXX_NoUnkeyedLiteral struct{}                    `json:"-"`
+	XXX_sizecache        int32                       `json:"-"`
+}
+
+func (m *CheckpointContainerRequest) Reset()      { *m = CheckpointContainerRequest{} }
+func (*CheckpointContainerRequest) ProtoMessage() {}
+func (*CheckpointContainerRequest) Descriptor() ([]byte, []int) {
+	return fileDescriptor_00212fb1f9d3bf1c, []int{107}
+}
+func (m *CheckpointContainerRequest) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *CheckpointContainerRequest) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_CheckpointContainerRequest.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *CheckpointContainerRequest) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_CheckpointContainerRequest.Merge(m, src)
+}
+func (m *CheckpointContainerRequest) XXX_Size() int {
+	return m.Size()
+}
+func (m *CheckpointContainerRequest) XXX_DiscardUnknown() {
+	xxx_messageInfo_CheckpointContainerRequest.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_CheckpointContainerRequest proto.InternalMessageInfo
+
+func (m *CheckpointContainerRequest) GetContainerId() string {
+	if m != nil {
+		return m.ContainerId
+	}
+	return ""
+}
+
+func (m *CheckpointContainerRequest) GetOptions() *CheckpointContainerOptions {
+	if m != nil {
+		return m.Options
+	}
+	return nil
+}
+
+type CheckpointContainerResponse struct {
+	XXX_NoUnkeyedLiteral struct{} `json:"-"`
+	XXX_sizecache        int32    `json:"-"`
+}
+
+func (m *CheckpointContainerResponse) Reset()      { *m = CheckpointContainerResponse{} }
+func (*CheckpointContainerResponse) ProtoMessage() {}
+func (*CheckpointContainerResponse) Descriptor() ([]byte, []int) {
+	return fileDescriptor_00212fb1f9d3bf1c, []int{108}
+}
+func (m *CheckpointContainerResponse) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *CheckpointContainerResponse) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_CheckpointContainerResponse.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *CheckpointContainerResponse) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_CheckpointContainerResponse.Merge(m, src)
+}
+func (m *CheckpointContainerResponse) XXX_Size() int {
+	return m.Size()
+}
+func (m *CheckpointContainerResponse) XXX_DiscardUnknown() {
+	xxx_messageInfo_CheckpointContainerResponse.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_CheckpointContainerResponse proto.InternalMessageInfo
+
+type RestoreContainerOptions struct {
```

Comment:
How about adding a checkpoint-dir option for a custom checkpoint storage directory?